buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/java/AvailableJavaInstallations.kt

```diff
@@ -79,35 +73,20 @@ open class AvailableJavaInstallations(private val project: Project, private val
     private
     fun determineJavaInstallation(propertyName: String): JavaInstallation {
```

Comment:
So this simplifies to the following, right:
```
fun determineJavaInstallation(propertyName: String): JavaInstallation =
    when (resolveJavaHomePath(propertyName)) {
        null -> currentJavaInstallation
        else -> detectJavaInstallation(resolvedJavaHome)
    }
```

Reply:
Unfortunately I prefer not to.

First `resolveJavaHomePath(propertyName)` should be repeated because it appears twice:

```
fun determineJavaInstallation(propertyName: String): JavaInstallation =
    when (resolveJavaHomePath(propertyName)) {
        null -> currentJavaInstallation
        else -> detectJavaInstallation(resolveJavaHomePath(propertyName))
    }
```

But this can't compile, we need to do more:

```
fun determineJavaInstallation(propertyName: String): JavaInstallation =
    when (resolveJavaHomePath(propertyName)) {
        null -> currentJavaInstallation
        else -> detectJavaInstallation(resolveJavaHomePath(propertyName)!!)
    }
```

I think this is not better than the current way...